ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -147,14 +151,18 @@ export default function TokenList({
             currencyRates,
           });
 
-          // Append processed token with balance and fiat amount
-          tokensWithBalance.push({
-            ...token,
-            balance,
-            tokenFiatAmount,
-            chainId,
-            string: String(balance),
-          });
+          // respect hide zero balance setting
+          // native tokens should still show zero balance regardless
+          // erc20s with zero balances should be hidden
+          if (!hideZeroBalanceTokens || token.isNative || balance !== '0') {
```

Comment:
Note: based on suggestion I think the `token.isNative` isn't a given and we'll need to add another condition here to only show a zero-balance native token if the "Current Network" is selected.  I think we should do that in a separate PR thought.

Reply:
It looks like @sahar-fehri added this condition added here: https://github.com/MetaMask/metamask-extension/pull/29073

Should we point and merge that PR into this one? They will need to be merged together eventually.

Reply:
I think they're simple enough to merge together.  They're changing the same if statement.